app/code/Magento/InventorySourceSelectionApi/Model/Request/LatLngRequestInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySourceSelectionApi\Model\Request;
+
+/**
+ * DTO for latitude and longitude request
+ *
+ * @api
+ */
+interface LatLngRequestInterface
+{
+    /**
+     * Get latitude
+     *
+     * @return float
+     */
+    public function getLat(): float;
+
+    /**
+     * Get longitude
+     *
+     * @return float
+     */
+    public function getLng(): float;
+
+    /**
+     * Get as string
+     *
+     * @return string
+     */
+    public function getAsString(): string;
```

Comment:
you are mixing data and presentation logic here which is not very good, as it's better to keep the separation of concerns principle.

Thus `getAsString` method should be moved to a dedicated service class

Reply:
Definetely right.